compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -1227,7 +1227,18 @@ static LogicalResult setDefaultGenericOpRootConfig(
                                  parallelTileSizes, reductionTileSizes);
 
   TileSizesListType tileSizes;
-  tileSizes.push_back(flowTileSizes);
+  SmallVector<int64_t> updatedFlowTileSizes;
+  bool foundNonZero = false;
+  for (auto val : flowTileSizes) {
+    if (val == 0) {
+      updatedFlowTileSizes.push_back(foundNonZero || tileSizes.size() <= 3 ? 0
+                                                                           : 1);
```

Comment:
`tileSizes.size() <= 3 ? 0 : 1` -> `tileSizes.size() > 3`?

Reply:
It's just annoying to rely on implicit `bool` to `int` conversions - one can get burnt by it easily when it doesn't do the right thing. I would rather be explicit to make it easier to read and write. Clang should optimize this anyway.

Reply:
Sorry, I actually read this as `(foundNonZero) || tileSizes.size() <= 3 ? 0 : 1;`. Agreed, whatever you can do to make it more explicit would help readability.

Reply:
I added parenthesis